tests/unit/test_config.py

```diff
@@ -28,13 +28,74 @@ def test_config_good_dict(good_config_dict):
     assert Config.from_config(config_dict=good_config_dict) == good_config_dict
 
 
-def test_config_missing_values(config_defaults):
+def test_config_extra_top_level_value(good_config_dict):
     """
-    GIVEN a json file containing an incomplete definition of a Bugjira config
+    GIVEN a dict containing a Bugjira config with an extra bugzilla setting
+    WHEN we call Config.from_config using the dict as the config_dict
+    THEN a ValidationError is raised with the appropriate error message
+    """
+    good_config_dict['foo'] = 'bar'
+    with pytest.raises(ValidationError,
+                       match="extra fields not permitted"):
+        Config.from_config(config_dict=good_config_dict)
+
+
+def test_config_extra_bugzilla_value(good_config_dict):
+    """
+    GIVEN a dict containing a Bugjira config with an extra bugzilla setting
+    WHEN we call Config.from_config using the dict as the config_dict
+    THEN a ValidationError is raised with the appropriate error message
+    """
+    good_config_dict['bugzilla']['foo'] = 'bar'
+    with pytest.raises(ValidationError,
+                       match="extra fields not permitted"):
+        Config.from_config(config_dict=good_config_dict)
+
+
+def test_config_extra_jira_value(good_config_dict):
+    """
+    GIVEN a dict containing a Bugjira config with an extra jira setting
+    WHEN we call Config.from_config using the dict as the config_dict
+    THEN a ValidationError is raised with the appropriate error message
+    """
+    good_config_dict['jira']['foo'] = 'bar'
+    with pytest.raises(ValidationError,
+                       match="extra fields not permitted"):
+        Config.from_config(config_dict=good_config_dict)
+
+
+def test_config_missing_value(good_config_dict):
+    """
+    GIVEN a dict containing a Bugjira config with one missing key
+    WHEN we call Config.from_config using the dict as the config_dict
+    THEN a ValidationError is raised with the appropriate error message
+    """
+    good_config_dict['jira'].pop('token_auth')
+    with pytest.raises(ValidationError,
+                       match="field required"):
```

Comment:
Question just for my own learning, will the field name be in the error message in that case?

Reply:
Good question. The ValidationError has lots of detail on the issues encountered during validation, so I'm sending up a revision that leverages this fact. Thanks for pressing for more detail on this.